### PR TITLE
simplify mac mkldnn build

### PR DESCRIPTION
**Pull request by azai91**
## Description ##
mxnet-mkldnn on mac requires that openmp be enabled (else performance is degraded). There are currently instructions specifying that the Makefile needs to be modified to address this; however, that is not a good user experience. 

the issue is that mac by default does not include openmp by default and therefore a guard was added to prevent users from trying to build with this. the developer should instead not build with openmp enabled on mac and the Makefile should not attempt to catch this (the user may falsely believe they are using openmp when they are not).

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Update MKLDNN readme docs
- [x] remove guard that prevents openmp from being enabled on mac

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by vrakesh** [comment]
@mxnet-label-bot [pr-awaiting-review]

**Comment by xinyu-intel** [comment]
Hi @azai91 , you can try to use llvm from `brew install llvm` instead of apple llvm built in xcode to enable OPENMP on both mxnet and mkldnn level. And I found this llvm also supports OPENCV well compared with `brew install gcc`. Details are as below:
```
brew install llvm
# .bash_profile
export LIBRARY_PATH=/usr/local/Cellar/llvm/6.0.1/lib/
# config.mk and mkldnn.mk
CC=/usr/local/Cellar/llvm/6.0.1/bin/clang CXX=/usr/local/Cellar/llvm/6.0.1/bin/clang++
```
And I get better performance on our local iMac Pro (Xeon W 8 cores with AVX-512):
```
INFO:root:network: vgg-16
INFO:root:device: cpu(0)
INFO:root:batch size  1, image/sec: 20.913986
INFO:root:batch size 16, image/sec: 24.273071
INFO:root:batch size 32, image/sec: 24.704907
INFO:root:batch size 64, image/sec: 24.946080
INFO:root:batch size 128, image/sec: 25.074148
INFO:root:batch size 256, image/sec: 25.262900
INFO:root:network: inception-v3
INFO:root:device: cpu(0)
INFO:root:batch size  1, image/sec: 41.431404
INFO:root:batch size 16, image/sec: 54.312317
INFO:root:batch size 32, image/sec: 54.604119
INFO:root:batch size 64, image/sec: 54.395680
INFO:root:batch size 128, image/sec: 54.410785
INFO:root:batch size 256, image/sec: 54.614424
INFO:root:network: inception-v4
INFO:root:device: cpu(0)
INFO:root:batch size  1, image/sec: 20.715221
INFO:root:batch size 16, image/sec: 26.249734
INFO:root:batch size 32, image/sec: 26.197659
INFO:root:batch size 64, image/sec: 26.161530
INFO:root:batch size 128, image/sec: 26.247461
INFO:root:batch size 256, image/sec: 26.313875
INFO:root:network: resnet-50
INFO:root:device: cpu(0)
INFO:root:batch size  1, image/sec: 40.505148
[23:07:34] src/operator/nn/mkldnn/mkldnn_base.cc:74: Allocate 51380224 bytes with malloc directly
INFO:root:batch size 16, image/sec: 41.431191
INFO:root:batch size 32, image/sec: 40.563484
INFO:root:batch size 64, image/sec: 37.070624
INFO:root:batch size 128, image/sec: 37.709096
INFO:root:batch size 256, image/sec: 37.448483
INFO:root:network: resnetv1-50
INFO:root:device: cpu(0)
INFO:root:batch size  1, image/sec: 41.701090
INFO:root:batch size 16, image/sec: 43.132788
INFO:root:batch size 32, image/sec: 41.613291
INFO:root:batch size 64, image/sec: 38.133290
INFO:root:batch size 128, image/sec: 38.839577
INFO:root:batch size 256, image/sec: 38.853521
INFO:root:network: mobilenet
INFO:root:device: cpu(0)
INFO:root:batch size  1, image/sec: 200.916080
INFO:root:batch size 16, image/sec: 287.614019
INFO:root:batch size 32, image/sec: 277.838051
INFO:root:batch size 64, image/sec: 274.474078
INFO:root:batch size 128, image/sec: 273.622323
INFO:root:batch size 256, image/sec: 273.445636
```

**Comment by lupesko** [comment]
@azai91 is there any documentation explaining to users how to install OpenMP so they can build the optimized version? if not - I suggest you consider adding it.

**Comment by pengzhao-intel** [comment]
@lupesko we have made up a doc in https://github.com/apache/incubator-mxnet/blob/master/MKLDNN_README.md

**Comment by sandeep-krishnamurthy** [comment]
@azai91 - what are the next steps here?

**Comment by azai91** [comment]
Try to find build optimal build instructions. Current ones provide cause a slowdown.

**Comment by azai91** [comment]
@xinyu-intel 
I get the following when running on resnet-50 with MKLDNN
```
INFO:root:batch size  1, dtype float32, images/sec: 16.791932
INFO:root:batch size  2, dtype float32, images/sec: 16.436919
INFO:root:batch size  4, dtype float32, images/sec: 15.737164
INFO:root:batch size  8, dtype float32, images/sec: 15.317152
INFO:root:batch size 16, dtype float32, images/sec: 11.681419
INFO:root:batch size 32, dtype float32, images/sec: 14.112597
```
vs without 
```
INFO:root:batch size  1, dtype float32, images/sec: 12.369148
INFO:root:batch size  2, dtype float32, images/sec: 15.216185
INFO:root:batch size  4, dtype float32, images/sec: 14.709387
INFO:root:batch size  8, dtype float32, images/sec: 14.508584
INFO:root:batch size 16, dtype float32, images/sec: 13.949826
INFO:root:batch size 32, dtype float32, images/sec: 13.240351
```

**Comment by azai91** [comment]
Benchmarking with mobilenet is better since the architecture is smaller:
with MKLDNN
```
INFO:root:batch size  1, dtype float32, images/sec: 107.649766
INFO:root:batch size  2, dtype float32, images/sec: 67.138190
INFO:root:batch size  4, dtype float32, images/sec: 115.700230
INFO:root:batch size  8, dtype float32, images/sec: 115.093671
INFO:root:batch size 16, dtype float32, images/sec: 116.750479
INFO:root:batch size 32, dtype float32, images/sec: 107.679933
```
w/o
```
INFO:root:batch size  1, dtype float32, images/sec: 39.162575
INFO:root:batch size  2, dtype float32, images/sec: 41.462039
INFO:root:batch size  4, dtype float32, images/sec: 42.517774
INFO:root:batch size  8, dtype float32, images/sec: 42.316606
INFO:root:batch size 16, dtype float32, images/sec: 40.210146
INFO:root:batch size 32, dtype float32, images/sec: 38.261067
```

**Comment by azai91** [comment]
updated the docs to support using llvm. @pengzhao-intel can you review this and if it looks good we can merge.

**Comment by azai91** [comment]
@mseth10

File: docs/install/osx_setup.md [hunk 1]
```diff
@@ -65,6 +65,10 @@ Install the dependencies, required for MXNet, with the following commands:
 	brew install openblas
 	brew tap homebrew/core
 	brew install opencv
+	
+	# If building with MKLDNN
+    brew install llvm
```
Comment (zheng-da) [inline docs/install/osx_setup.md]: indent

**Pull request merged**